src/libs/OptionsListUtils.ts

```diff
@@ -1625,131 +1669,142 @@ function getValidReports(reports: OptionList['reports'], config: GetValidReports
 
-    for (let i = 0; i < reports.length; i++) {
-        // eslint-disable-next-line rulesdir/prefer-at
-        const option = reports[i];
-        const report = option.item;
-        const chatReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report.chatReportID}`];
-        const doesReportHaveViolations = shouldDisplayViolationsRBRInLHN(report, transactionViolations);
+    const shouldBeInOptionList = shouldReportBeInOptionList({
+        report,
+        chatReport,
+        currentReportId: topmostReportId,
+        betas,
+        doesReportHaveViolations,
+        isInFocusMode: false,
+        excludeEmptyChats: false,
+        includeSelfDM,
+        login: option.login,
+        includeDomainEmail,
+        isReportArchived: !!option.private_isArchived,
+    });
 
-        const shouldBeInOptionList = shouldReportBeInOptionList({
-            report,
-            chatReport,
-            currentReportId: topmostReportId,
-            betas,
-            doesReportHaveViolations,
-            isInFocusMode: false,
-            excludeEmptyChats: false,
-            includeSelfDM,
-            login: option.login,
-            includeDomainEmail,
-            isReportArchived: !!option.private_isArchived,
-        });
+    if (!shouldBeInOptionList) {
+        return false;
+    }
 
-        if (!shouldBeInOptionList) {
-            continue;
-        }
+    const isThread = option.isThread;
+    const isTaskReport = option.isTaskReport;
+    const isPolicyExpenseChat = option.isPolicyExpenseChat;
+    const isMoneyRequestReport = option.isMoneyRequestReport;
+    const isSelfDM = option.isSelfDM;
+    const isChatRoom = option.isChatRoom;
+    const accountIDs = getParticipantsAccountIDsForDisplay(report);
 
-        const isThread = option.isThread;
-        const isTaskReport = option.isTaskReport;
-        const isPolicyExpenseChat = option.isPolicyExpenseChat;
-        const isMoneyRequestReport = option.isMoneyRequestReport;
-        const isSelfDM = option.isSelfDM;
-        const isChatRoom = option.isChatRoom;
-        const accountIDs = getParticipantsAccountIDsForDisplay(report);
+    if (excludeNonAdminWorkspaces && !isPolicyAdmin(option.policyID, policies)) {
+        return false;
+    }
 
-        if (excludeNonAdminWorkspaces && !isPolicyAdmin(option.policyID, policies)) {
-            continue;
-        }
+    if (isPolicyExpenseChat && report?.isOwnPolicyExpenseChat && !includeOwnedWorkspaceChats) {
+        return false;
+    }
+    // When passing includeP2P false we are trying to hide features from users that are not ready for P2P and limited to expense chats only.
+    if (!includeP2P && !isPolicyExpenseChat) {
+        return false;
+    }
 
-        if (isPolicyExpenseChat && report.isOwnPolicyExpenseChat && !includeOwnedWorkspaceChats) {
-            continue;
-        }
+    if (isSelfDM && !includeSelfDM) {
```

Comment:
NAB but heuristically it might make sense to check this after `isThread`, `isTaskReport`, `isMoneyRequestReport`, etc... because those all could be true for `N` reports, while this will only be true for `1` report. For for all those N reports you're doing this check first unnecessarily.